extension/src/background/messageListener/freighterApiMessageListener.ts

```diff
@@ -74,9 +75,10 @@ export const freighterApiMessageListener = (
 
   const submitTransaction = async () => {
     const { transactionXdr } = request;
+    const { network, networkUrl } = getNetworkDetails(getIsTestnet());
```

Comment:
this is basically how it works now - figure out if we're on testnet based on localStorage, and then get the appropriate config